app/models/viewmodels/notificationspage/converters/ControlResponseConverter.scala

```diff
@@ -54,27 +54,17 @@ class ControlResponseConverter @Inject()(decoder: Decoder, dateTimeFormatter: Da
     }
 
   private def buildActionCodeExplanation(actionCode: String)(implicit messages: Messages): Option[String] =
-    decoder.actionCode(actionCode).map(code => paragraph(messages(code.contentKey)))
-
-  private def buildErrorExplanation(errorCode: String)(implicit messages: Messages): Option[String] = {
-    val isChiefError = decoder.chiefErrorCode(errorCode).isDefined
-
-    if (isChiefError) buildCHIEFErrorExplanation(errorCode)
-    else buildILEErrorExplanation(errorCode)
-  }
+    decoder.actionCode(actionCode).map(code => paragraph(messages(code.messageKey)))
 
   // TODO move logging for missing error codes to backend
-  private def buildILEErrorExplanation(errorCode: String)(implicit messages: Messages): Option[String] =
-    decoder.ileErrorCode(errorCode).map(error => paragraph(messages(error.contentKey))) match {
+  private def buildErrorExplanation(errorCode: String)(implicit messages: Messages): Option[String] =
+    decoder.error(errorCode).map(error => paragraph(messages(error.messageKey))) match {
       case None =>
         logger.info(s"Received inventoryLinkingControlResponse with unknown error code: $errorCode")
         None
       case error => error
     }
 
-  private def buildCHIEFErrorExplanation(errorCode: String): Option[String] =
-    decoder.chiefErrorCode(errorCode).map(error => paragraph(error.description))
-
   private val paragraph: String => String = (text: String) => s"<p>$text</p>"
```

Comment:
Not sure about building HTML outside the views, but I guess you've not changed that.  I'd just have this converter return messages (or message keys).  But it's not important for this PR

Reply:
I agree, but I feel it is simpler to do it this way for now since the design is changing a lot in this area. And sometimes we need to use paragraphs, and sometimes we need lists...